### PR TITLE
novops: init at v0.12.0

### PR DESCRIPTION
**Pull request by PierreBeucher**
## Description of changes

[Novops](https://github.com/PierreBeucher/novops) is a secret and config manager for development and CI environment. It loads secrets and configs as files and environment variable in current environment. Secrets are securely loaded (as tmpfs files or directly in process memory) and only kept for as long as they are needed.

Novops works on x86_64 / aarch64 and Linux / Darwin, though I was not able to test specifically aarch64 Nix build (non-Nix build works). 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/no/novops/package.nix
Comment (sikmir) [inline pkgs/by-name/no/novops/package.nix:17]:
```suggestion
    repo = "novops";
```

**Comment by jnsgruk** [comment]
Looking good - can you please rebase to just two commits:

```
novops: init at 0.12.0
maintainers: add pbeucher
```

**Comment by teto** [comment]
in other ecosystems we've done a lot to avoid committing huge cargofiles such as pkgs/by-name/no/novops/Cargo.lock . I am not familiar with Is there any way to avoid that (like substitute paths in the lockfile instead) ? Maybe we can wait for https://github.com/jmgilman/vaultrs/pull/58 to get merged (maintainer self requested review last week).

**Comment by PierreBeucher** [comment]
Indeed this lockfile should be temporary, it contains a depency on direct-link to Github repo for `vaultrs` which forced me to include it here. There's a new maintainer on vaultrs so hopefully it'll get merged soon

**Comment by PierreBeucher** [comment]
Waiting for next release of https://github.com/jmgilman/vaultrs to remove custom lock file and rebase, then this PR will be ready to be merged :)

**Comment by PierreBeucher** [comment]
Rebased and removed cargo.lock following vaultrs release, all good :)

**Comment by teto** [comment]
Result of `nixpkgs-review pr 290658` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>novops</li>
  </ul>
</details>

**Review by teto — APPROVED** [review]

**Comment by teto** [comment]
I quite like this tool. Merging once CI passes ;)

**Pull request merged**
